python/cudf/cudf/core/index.py

```diff
@@ -1467,6 +1467,136 @@ def __cuda_array_interface__(self):
         return self._values.__cuda_array_interface__
 
 
+class Int64Index(GenericIndex):
+    """Immutable, ordered and sliceable sequence of integer labels.
+    The basic object storing row labels for all cuDF objects.
+    Int64Index is a special case of Index with purely integer labels.
+
+    Parameters:
+    -----------
+    data : array-like (1-dimensional)
+    dtype : NumPy dtype (default: int64)
+            If dtype is None, we find the dtype that best fits the data.
+    copy : bool
+        Make a copy of input data.
+    name : object
+        Name to be stored in the index.
+
+    Returns
+    -------
+    Index
+
+    Examples
+    --------
+    >>> import cudf
+    >>> cudf.Int64Index([1, 2, 3, 4], name="a")
+    Int64Index([1, 2, 3, 4], dtype='int64', name='a')
+
+    >>> cudf.Int64Index([1, 2, 3, 4], dtype="int16", name="a")
+    Int64Index([1, 2, 3, 4], dtype='int64', name='a')
+    """
+
+    def __new__(cls, data=None, dtype="int64", copy=False, name=None):
+
+        out = Frame().__new__(cls)
+        if copy is True and data is not None:
+            data = data.copy()
+
+        kwargs = _setdefault_name(data, name=name)
+
+        data = column.as_column(data, dtype=dtype)
+
+        out._initialize(data, **kwargs)
+
+        return out
```

Comment:
If `dtype` is say `int8` here this doesn't have the expected behavior of eventually returning a `int64` typed object. I think we need to cast to `int64` at the end regardless.

Reply:
Not, it will return `int64` even though you pass `int8`.

Reply:
Where does the cast to `int64` happen? I'm not seeing it.

Reply:
Oh, I was misled by the `to_pandas`, will update this.

Reply:
In that case, whatever be the dtype, we will just use `int64` while converting.